src/TrinaryLogic.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace OndraM\CiDetector;
+
+/**
+ * @see https://en.wikipedia.org/wiki/Three-valued_logic
+ * @see https://github.com/phpstan/phpstan-src/blob/cedc99f51f7b8d815036e983166d7cb51ab46734/src/TrinaryLogic.php
+ *
+ * @internal
+ */
+final class TrinaryLogic
+{
+    private const YES = 1;
+    private const MAYBE = 0;
+    private const NO = -1;
+
+    /** @var self[] */
+    private static $registry = [];
+
+    /** @var int */
+    private $value;
+
+    private function __construct(int $value)
+    {
+        $this->value = $value;
+    }
+
+    public static function createMaybe(): self
+    {
+        return self::create(self::MAYBE);
+    }
+
+    public static function createFromBoolean(bool $value): self
+    {
+        return self::create($value ? self::YES : self::NO);
+    }
+
+    private static function create(int $value): self
+    {
+        self::$registry[$value] = self::$registry[$value] ?? new self($value);
+
+        return self::$registry[$value];
+    }
+
+    public function yes(): bool
+    {
+        return $this->value === self::YES;
+    }
+
+    public function maybe(): bool
```

Comment:
I'm missing documentation for the public methods, especially this `maybe()` and `describe()`, so that you know in IDE what `maybe()` means if you use the method (without inspecting its source code).

Could you please add them?

Reply:
Do you have any suggestion ? :)

Reply:
Something like:
Maybe(): Return true if its not known for sure whether the value is true or false
Describe(): Return string representation of the value. "Yes" when the value is true, "No" when its false, "Maybe" when its not known for sure whether its true or false.